oss-internship-2020/curl/examples/example1.cc

```diff
@@ -12,67 +12,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <libgen.h>
-#include <syscall.h>
 #include <iostream>
+
 #include "curl_sapi.sapi.h"
 #include "sandboxed_api/util/flag.h"
 
-/*
-int main() {
-    CURL *curl = curl_easy_init();
-    curl_easy_setopt(curl, CURLOPT_VERBOSE, 1l);
-    curl_easy_setopt(curl, CURLOPT_URL, "http://example.com");
-    curl_easy_perform(curl);
-    curl_easy_cleanup(curl);
-}
-*/
-
 class CurlApiSandboxEx1 : public CurlSandbox {
-
-  std::unique_ptr<sandbox2::Policy> ModifyPolicy(
-      sandbox2::PolicyBuilder*) override {
-
-    return sandbox2::PolicyBuilder()
-        .DangerDefaultAllowAll()
-        .AllowUnrestrictedNetworking()
-        .AddDirectory("/lib")
-        .BuildOrDie();
-  }
-
+  private:
+    std::unique_ptr<sandbox2::Policy> ModifyPolicy( 
+        sandbox2::PolicyBuilder*) override {
+      // Return a new policy
+      return sandbox2::PolicyBuilder()
+          .DangerDefaultAllowAll()
+          .AllowUnrestrictedNetworking()
+          .AddDirectory("/lib")
+          .BuildOrDie();
+    }
 };
 
+// GET http://example.com
 int main(int argc, char* argv[]) {
 
-  CurlApiSandboxEx1 sandbox;
-  
-  auto status_sandbox_init = sandbox.Init();
-  assert(status_sandbox_init.ok());
-
+  CurlApiSandboxEx1 sandbox; 
+  absl::Status status = sandbox.Init();
+  assert(status.ok());
   CurlApi api(&sandbox);
 
-  auto status_init = api.curl_easy_init();
-  assert(status_init.ok());
+  sapi::StatusOr<CURL*> status_or_culrptr = api.curl_easy_init();
+  assert(status.ok());
 
-  ::sapi::v::RemotePtr curl(status_init.value());
-  assert(curl.GetValue());
+  sapi::v::RemotePtr curl(status_or_culrptr.value());
```

Comment:
FYI: `status_or_culrptr.value()` is equivalent to `*status_or_curl`, if you apply the comment above.

In example code we should stay explicit, though, as the deref operator may not be something users might be accomodated with.